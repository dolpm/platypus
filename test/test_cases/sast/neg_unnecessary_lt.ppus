pipe inner |'a| [rock: &'a int] |> &'a int {   
    |> rock;
}

pipe outer |'c, 'd| [mic: &'c int, treble : &'d int] |> &'d int {       
    &int shinola = inner <| [mic];
    |> shinola;
}

pipe main [] |> unit {
    int coffee = 0;
    {
        int black = 1;
        &int just_like_my_metal = outer <| [&coffee, &black];
    }

    |> ();
}