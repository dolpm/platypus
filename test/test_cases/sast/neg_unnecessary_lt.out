Fatal error: exception Failure("variable treble will not be returned from outer and thus its lifetime isn't required")
